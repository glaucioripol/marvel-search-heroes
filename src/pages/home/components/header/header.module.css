.header {
  display: flex;
  justify-content: center;
  margin-block: 24px;
}

.header .header__logo {
  width: 326px;
  height: 120px;
}
