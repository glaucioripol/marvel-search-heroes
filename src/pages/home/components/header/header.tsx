import styles from "./header.module.css";

export function Header() {
  return (
    <header className={styles.header}>
      <img
        className={styles.header__logo}
        src="/assets/logo.svg"
        alt="Texto escrito 'Marvel Search heroes', a parte escrito marvel estÃ¡ em vermelho e o restante em cinza escuro."
        loading="lazy"
        data-testid="header-logo"
      />
    </header>
  );
}
