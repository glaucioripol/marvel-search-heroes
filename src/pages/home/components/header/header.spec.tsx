import {
  defaultRender,
  screen,
  beforeAll,
  describe,
  expect,
  it,
} from "@/__tests__";
import { Header } from "./header";

describe("Header component", () => {
  let component: ReturnType<typeof defaultRender>;
  beforeAll(() => {
    component = defaultRender(<Header />);
  });

  it("should render without crashing", () => {
    expect(screen.getByTestId("header-logo")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
