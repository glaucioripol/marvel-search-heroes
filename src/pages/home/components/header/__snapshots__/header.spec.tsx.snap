// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > should match snapshot 1`] = `<DocumentFragment />`;
